dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusPTX.java

```diff
@@ -0,0 +1,311 @@
+        IValue constant = new IValue(new BigInteger(ctx.constant().getText()), getArchPrecision());
+        String sem = ctx.sem().content;
+        String scope;
+        if (sem.equals(Tag.PTX.WEAK)) {
+            if (ctx.scope() != null) {
+                throw new ParsingException("Weak store instruction doesn't need scope: " + ctx.scope().content);
+            }
+            scope = Tag.PTX.SYS;
+        } else if (sem.equals(Tag.PTX.REL) || sem.equals(Tag.PTX.RLX)) {
+            scope = ctx.scope().content;
+        } else {
+            throw new ParsingException("Store instruction doesn't support sem: " + ctx.sem().content);
+        }
+        Store store = EventFactory.PTX.newTaggedStore(object, constant, sem, scope);
+        store.addFilters(ctx.store().storeProxy);
+        store.addFilters(Tag.PTX.CON);
+        return programBuilder.addScopedChild(mainThread, store);
+    }
+
+    @Override
+    public Object visitStoreRegister(LitmusPTXParser.StoreRegisterContext ctx) {
+        MemoryObject object = programBuilder.getOrNewObject(ctx.location().getText());
+        Register register = programBuilder.getOrCreateRegister(mainThread, ctx.register().getText(), getArchPrecision());
+        String sem = ctx.sem().content;
+        String scope;
+        if (sem.equals(Tag.PTX.WEAK)) {
+            if (ctx.scope() != null) {
+                throw new ParsingException("Weak store instruction doesn't need scope: " + ctx.scope().content);
+            }
+            scope = Tag.PTX.SYS;
+        } else if (sem.equals(Tag.PTX.REL) || sem.equals(Tag.PTX.RLX)) {
+            scope = ctx.scope().content;
+        } else {
+            throw new ParsingException("Store instruction doesn't support sem: " + ctx.sem().content);
+        }
+        Store store = EventFactory.PTX.newTaggedStore(object, register, sem, scope);
+        store.addFilters(ctx.store().storeProxy);
+        return programBuilder.addScopedChild(mainThread, store);
+    }
+
+    @Override
+    public Object visitLoadConstant(LitmusPTXParser.LoadConstantContext ctx) {
+        Register register = programBuilder.getOrCreateRegister(mainThread, ctx.register().getText(), getArchPrecision());
+        IValue constant = new IValue(new BigInteger(ctx.constant().getText()), getArchPrecision());
+        String sem = ctx.sem().content;
+        String scope;
+        if (sem.equals(Tag.PTX.WEAK)) {
+            if (ctx.scope() != null) {
+                throw new ParsingException("Weak load instruction doesn't need scope: " + ctx.scope().content);
+            }
+            scope = Tag.PTX.SYS;
+        } else if (sem.equals(Tag.PTX.ACQ) || sem.equals(Tag.PTX.RLX)) {
+            scope = ctx.scope().content;
+        } else {
+            throw new ParsingException("Load instruction doesn't support sem: " + ctx.sem().content);
+        }
+        return programBuilder.addChild(mainThread, EventFactory.newLocal(register, constant));
+    }
```

Comment:
The sem and scope are only used for validating the instruction. This local type instruction is following other memory models. Should we keep it or remove this kind of instruction from PTX?

Reply:
You need the visitor because you need to add the `Local` event to the program. But everything you are computing in lines 176-187 is never used, thus you can completely remove those lines and nothing would change,

Reply:
Removed